src/radical/pilot/agent/executing/popen.py

```diff
@@ -102,16 +102,10 @@ def work(self, tasks):
                 self._prof.prof('task_start', uid=task['uid'])
                 self._handle_task(task)
 
-            except Exception:
-                # append the startup error to the tasks stderr.  This is
-                # not completely correct (as this text is not produced
-                # by the task), but it seems the most intuitive way to
-                # communicate that error to the application/user.
+            except Exception as e:
                 self._log.exception("error running Task")
-                if task['stderr'] is None:
-                    task['stderr'] = ''
-                task['stderr'] += '\nPilot cannot start task:\n'
-                task['stderr'] += '\n'.join(ru.get_exception_trace())
+                task['exception']        = repr(e)
+                task['exception_detail'] = '\n'.join(ru.get_exception_trace())
```

Comment:
Should the similar update be applied to the Flux executor? ([here](https://github.com/radical-cybertools/radical.pilot/blob/dfb4a2ce38de27c591da43eaaae110cc92ba9b15/src/radical/pilot/agent/executing/flux.py#L264-L267))

Reply:
Fair - that's done now.